### PR TITLE
Allow adding custom border widths and colours

### PR DESCRIPTION
**Pull request by archferns**
This commit adds two new parameters to the add_border method, to be able to specify border width and colour.

### CODE REVIEW

**Comment by sakovias** [comment]
Hi @archferns. Great work! Thanks for adding this feature.

I have some thoughts on the signature of `Worksheet#add_border` method. I'm thinking of making it more consistent with the original `axlsx` gem:
```
add_border(<cell_range>, <args>)
```
where `args` are either edges as it used to be (to not break the clients) or a hash of parameters. These both should be valid calls:
```
sheet.add_border 'B2:D5', [:right]
sheet.add_border 'B2:D5', { style: :thick, color: "FFFF0000", edges: [:top, :bottom] }
```
I think this creates an opportunity to have a cleaner API going forward.

What do you think?

File: README.md [hunk 1]
```diff
@@ -39,6 +39,9 @@ You can also add borders to ranges of cells:
 ```ruby
 sheet.add_border 'B2:D5'
 sheet.add_border 'B2:D5', [:right]
+sheet.add_border 'B2:D5', [:right], :thick
+border_color = 'FF0000'
```
Comment (sakovias) [inline README.md]: I'd just move it to the next line in explicit way
```ruby
sheet.add_border 'B2:D5', [:right], :medium, 'FF0000'
```

**Comment by sakovias** [comment]
We also need to test what happens when borders with different styles overlap. For example, we apply a red thick border to cells `A1:A3` and a green medium border to cells `B1:B3`. The test should show that the number of unique styles is correct and the looks are consistent with `axlsx` gem.

Feel free to do jump in or let me know if you'd rather have me do it.

**Comment by archferns** [comment]
Yes, it is definitely better to pass a hash instead.
If its convenient in your schedule, please do go ahead, or else, I can look into it in a few days time.

**Comment by sakovias** [comment]
Sounds good. I'll make a change some time later this week.

**Comment by sakovias** [comment]
@archferns Made changes in line with what we've discussed. Will add some tests in a bit. Feel free to take a look: https://github.com/sakovias/axlsx_styler/commits/pr-custom-border

**Comment by archferns** [comment]
@sakovias , yes, that is perfect!
By the way, I can confirm that the overlapping of borders is an issue, since I encountered it when I tried applying multiple border widths/colours.

**Comment by archferns** [comment]
I suppose its time for me to familiarize myself with the `axlsx` documentation, I will try to fix it a little later.

**Comment by sakovias** [comment]
There was a bug with border width which I fixed.

I've also added a couple of tests. If I understand it correctly, `axlsx` doesn't allow to apply different border styles to different cell edges (e.g. top - thick, red; bottom - thin, green). We just need to remember that styles are being merged in `axlsx_styler` and if duplicate styles are applied the last style wins. Take a looks at these tests:
https://github.com/sakovias/axlsx_styler/commit/cfa85584851e2bd493c179cd7af8db918ca40302

Let me know if there's something you can do with `axlsx`, but not `axlsx_styler`. We should be able to fix it.

**Comment by sakovias** [comment]
Let me know if you have problems with applying overlapping borders. I can take a look if you give me some details.

**Comment by sakovias** [comment]
@archferns Please review https://github.com/sakovias/axlsx_styler/pull/4

Sorry, I didn't mean to open a new pull request. Just didn't take my time to figure out how to do this. You original work is included.

**Pull request closed without merging**